grammar.js

```diff
@@ -493,10 +493,57 @@ module.exports = grammar(C, {
       $.throw_statement,
     ),
 
+    condition_declaration: $ => seq(
+      '(',
+      $._declaration_specifiers,
+      field('declarator', $._declarator),
+      choice(
+        seq(
+          '=',
+          field('default_value', $._expression),
+        ),
+        $.initializer_list,
+      ),
+      ')',
+    ),
+
+    init_statement: $ => seq(
+      '(',
+      choice(
+        field('initializer', $.declaration),
+        seq(field('initializer', optional(choice($._expression, $.comma_expression))), ';'),
+      ),
+      field('condition', optional($._expression)),
+      ')',
+    ),
```

Comment:
Could `condition_declaration` and `init_statement` be unified? It seems like the contents of `condition_declaration` is mostly the same as `declaration`.

Reply:
@maxbrunsfeld, the `condition_declaration` isn't a `declaration` because it doesn't contain a final semi-colon.
So not sure to see how to make better here, any help would be appreciated.